### PR TITLE
Split rc_input_events

### PR DESCRIPTION
**Pull request by FrankvRaffe**
Split rc_input_events containing three functions into rc_input_event_crop, rc_input_event_amendment, and rc_input_events

## Summary by CodeRabbit

* **New Features**
  * Generates monthly carbon-input events for crops and amendments and precomputes inputs for simulations.

* **Changed**
  * Simplified interface: callers supply crop/amendment input tables; simulation horizon defaults to 50 years and inputs are auto-replicated; clay-content parameter removed.

* **Bug Fixes**
  * Corrected dependency typos.

* **Documentation**
  * README expanded and changelog (NEWS) updated.

* **Tests**
  * Added comprehensive unit tests for event generation, aggregation, replication, timing, and edge cases.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # rotsee UNRELEASED
+## Changed
```
Comment (gerardhros) [inline NEWS.md:2]: set version number to latest one and add date.

File: R/rc_input_event_amendment.R [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  checkmate::assert_numeric(dt$fr_eoc_p,lower = 0, upper = 250,len = nrow(dt))
+  checkmate::assert_integerish(dt$year,len = nrow(dt))
+  
+  # load internal tables
+  cc.crop <- as.data.table(rotsee::rc_crops)
+  cc.crop[,B_LU := paste0('nl_',crop_code)]
+  
+  # merge table with amendment c input with crop name
+  dt <- merge(dt, crops[,list(B_LU,year)], by='year')
+  dt <- merge(dt, cc.crop[,list(B_LU,B_LU_NAME=crop_name)],by='B_LU')
```
Comment (gerardhros) [inline R/rc_input_event_amendment.R:52]: when doing a left join, always add, all.x = TRUE. This ensures that all original rows of dt remain in the merge (otherwise they are automatically removed)

File: R/rc_input_event_amendment.R [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  checkmate::assert_numeric(dt$cin_tot,lower = 0, upper = 100000,len = nrow(dt))}
+  checkmate::assert_numeric(dt$cin_dpm,lower = 0, upper = 100000,len = nrow(dt))
+  checkmate::assert_numeric(dt$cin_rpm,lower = 0, upper = 100000,len = nrow(dt))
+  checkmate::assert_numeric(dt$fr_eoc_p,lower = 0, upper = 250,len = nrow(dt))
+  checkmate::assert_integerish(dt$year,len = nrow(dt))
+  
+  # load internal tables
+  cc.crop <- as.data.table(rotsee::rc_crops)
+  cc.crop[,B_LU := paste0('nl_',crop_code)]
+
```
Comment (gerardhros) [inline R/rc_input_event_amendment.R:49]: this conversion to dutch crop coding is temporary i assume

File: R/rc_input_event_amendment.R [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  dt <- merge(dt, crops[,list(B_LU,year)], by='year')
+  dt <- merge(dt, cc.crop[,list(B_LU,B_LU_NAME=crop_name)],by='B_LU')
+  
+  # add manure category depending on eoc-to-p ratio
+  # assuming that soil improving products (high ratio) are incorporated in autumn, and others in spring
+  dt[,p_cat := fifelse(fr_eoc_p > 20, 'autumn','spring')]
+  
+  # Sum C inputs over p_cat per year
+  cols <- c('cin_hum','cin_dpm','cin_rpm')
+  dt <- dt[,lapply(.SD,function(x) sum(x)),.SDcols = cols,by = c('year','p_cat','B_LU','B_LU_NAME')]
```
Comment (gerardhros) [inline R/rc_input_event_amendment.R:60]: why are you summing the inputs on annual level? the events output can have C inputs at any time, isn't?

Comment (gerardhros) [reply R/rc_input_event_amendment.R:60]: i see in the code below you asre setting all inputs into a fixed month depending on crop. if the input contains the events on montly level already, this "redistribution" of annual fluxes over the months is not needed isn't?

File: R/rc_input_event_amendment.R [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    dt2 <- copy(dt)[,tcf := 1]
+    
+    # when month is unknown, assume that month is April
+    dt2[is.na(month),month := 4]
+    
+  }
+  
+  # sum all inputs per crop and year
+  cols <- c('cin_hum','cin_dpm','cin_rpm')
+  dt2 <- dt2[,lapply(.SD,function(x) sum(x * tcf)),
```
Comment (gerardhros) [inline R/rc_input_event_amendment.R:109]: why sum them here when creating an event data.table?

File: R/rc_input_event_crop.R [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  
+  # get internal copies of tables
+  cc.crop <- as.data.table(rotsee::rc_crops)
+  cc.crop[,B_LU := paste0('nl_',crop_code)]
+  
+  # add crop names and catch crop characteristics to table
+  dt <- merge(dt,cc.crop,by = 'B_LU',all.x = TRUE)
+ 
+  # adjust crop rotation plan, and extend to months
+  # update input for mandatory catch crops (after maize and potato on sandy soils) and set never for grassland
```
Comment (gerardhros) [inline R/rc_input_event_crop.R:40]: this is NL ...

**Review by gerardhros — CHANGES_REQUESTED** [review]
see my questions added.

File: R/rc_input_event_amendment.R (hunk 3 above)
Comment (FrankvRaffe) [reply R/rc_input_event_amendment.R:49]: Correct, I will resolve the non-generic of crop and amendment management parts of rotsee in the branch "incorporating carboncastr"

File: R/rc_input_event_amendment.R (hunk 4 above)
Comment (FrankvRaffe) [reply R/rc_input_event_amendment.R:60]: I will leave it in for now as I will improve the specific code in the branch "incorporating carboncastr", but looking at it:

I believe you are correct. Either no monthly distribution is given as input (in which case total C input should be known and summing is unnecessary) or a monthly distributionis given, in which case this is also not necessary.

File: R/rc_input_event_amendment.R [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+    dt2[crflt == 1 & month == 7, tcf := 0.10]
+    dt2[crflt == 1 & month == 9, tcf := 0.20]
+    dt2[,crflt := NULL]
+    dt2[grepl('^nl_',B_LU) & grepl('gras',B_LU_NAME) & p_cat == 'autumn', tcf := fifelse(month==10,1,0)]
+    
+    # add timings for grassland not in the Netherlands
+    dt2[grepl('gras',B_LU_NAME) & !grepl('nl_',B_LU), tcf := fifelse(month %in% c(4,6,8),1/3,0)]
+    
+    # add timings for non grassland systems
+    dt2[!grepl('gras',B_LU_NAME) & p_cat == 'spring', tcf := fifelse(month == 4, 1, 0)]
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_event_amendment.R:85]: This will again be resolve in the branch "incorporating carboncastr". We can still discuss on exact implementation, but my current idea is that rotsee as a calculation core solely here relies on user inputs for when amendments are applied and at what dosages. As such, this whole block will be removed. We can consider creating a separate helper function where such distributions can be calculated for easier input generation.

File: R/rc_input_event_crop.R (hunk 6 above)
Comment (FrankvRaffe) [reply R/rc_input_event_crop.R:40]: Yes, will remove in branch "incorporating carboncastr" and instead let the user give these inputs to steer from relying on Dutch standards.

File: R/rc_input_events.R [hunk 8]
```diff
@@ -2,25 +2,26 @@
...
+  checkmate::assert_names(colnames(crops), must.include = c("time", "var", "value", "method"))
+  checkmate::assert_data_table(amendment, any.missing = FALSE)
+  checkmate::assert_names(colnames(amendment), must.include = c("time", "var", "value", "method"))
   
-  # estimate Carbon input via manure, compost and organic residues
-  event.man <- rc_input_event_amendment(crops = crops,amendment = amendment)
   
   # create event
-  rothc.event <- rbind(event.crop,event.man)
+  rothc.event <- rbind(crops,amendment)
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_events.R:24]: This is currently done from lines 32 to 49, indeed repeating events until the simyears period is covered and then ordering by time. Exact implementation will potentially change when switching from simyears to start_date to end_date.

**Review by SvenVw — APPROVED** [review]

File: R/rc_input_event_crop.R [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  checkmate::assert_integerish(crops$year, any.missing = FALSE, len = arg.length)
+  if (!"month" %in% names(crops)) crops[, month := NA_real_]
+  
+  
+  # make internal copy
+  dt <- copy(crops)
+  
+  # If month is not supplied, set to 9
+  dt[, month := as.integer(month)]
+  dt[is.na(month), month := 9]
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_event_crop.R]: @coderabbitai I added this in my latest commit

File: tests/testthat/test-rc_input_events.R [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+  
+  amendment <- data.table(
+    time = numeric(0),
+    var = character(0),
+    method = character(0), 
+    value = numeric(0)
+  )
+  
+  result <- rc_input_events(crops, amendment, simyears = 10)
+})
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply tests/testthat/test-rc_input_events.R:68]: @coderabbitai I added this in my latest commit
